### ISSUE TITLE
Refactor UIWidget to always use UILayoutComponent

### ISSUE THREAD

**Issue by Shulepov** [cocos2d/cocos2d-x#16129]
I'm am trying to look into new UI system again. And find it quit hard to follow.

UIWidget and UILayoutComponent have duplicate functionality. UIWidget can use UILayoutComponent and also can not use. In case it not uses UILayoutComponent - it duplicates almost the same code. Also the same things called differently for those two classes. May be in all cases should be used UILayoutComponent to remove some ambiguity?

**Comment by minggo** [comment]
@Shulepov will it break compatibility?

**Comment by Shulepov** [comment]
@minggo not sure, at first glance it shouldn't
